trinity/sync/full/chain.py

```diff
@@ -354,7 +357,38 @@ class FastChainSyncer(BaseHeaderChainSyncer):
         if not receipt_bundles:
             return tuple(), batch
 
-        receipts, trie_roots_and_data_dicts = zip(*receipt_bundles)
+        # First validate the receipts.
+        headers_by_root = {
```

Comment:
To distinguish between single header values here, and multiple receipts values in the next dict, maybe call this: `header_by_root`?